lychee-lib/src/utils/path.rs

```diff
@@ -42,14 +42,14 @@ fn dirname(src: &'_ Path) -> Option<&'_ Path> {
 /// Returns Ok(None) in case of an absolute local link without a `base_url`
 pub(crate) fn resolve(src: &Path, dst: &Path, base: &Option<Base>) -> Result<Option<PathBuf>> {
     let resolved = match dst {
-        relative if dst.is_relative() => {
+        relative if !dst.starts_with("/") => {
```

Comment:
this change is unrelated to the rest of the PR

I needed it because I'm running on Windows where a path only seems to be considered absolute if it contains a drive letter

Reply:
That's surprising, because this is how I understood the documentation of `Path` in the standard library:
It should handle Windows correctly, see [`Path::is_absolute`](https://doc.rust-lang.org/std/path/struct.Path.html#method.is_absolute).

(Note that `Path::is_relative` is simply implemented as `!Path::is_absolute`.)

Reply:
yeah, but I think(?) this condition should be about `abc` (relative) vs `/xyz` (absolute), e.g.

```
<html>
    <body>
        <a href="abc">home</a>
        <a href="/xyz">home</a>
    </body>
</html>
```

but `is_relative()` returns true for both `abc` and `/xyz` on Windows

I don't mind reverting it though, as I'll be running lychee via the docker image anyways

Reply:
Aha! Yeah, that makes sense. 

Since we'll no longer use `is_relative` from the standard library, I suggest we encapsulate that behavior in our own function and adding a few unit tests here to check our expectations. 



```rust
use std::path::Path;

/// Check if a given path is relative
///
/// Note that we don't use [`Path::is_relative`] here, as it returns true
/// for both `abc` and `/xyz` on Windows and `/xyz` is expected to be a relative
/// path as per our link checking logic
fn is_relative_path(path: &Path) -> bool {
    !is_absolute_path(path)
}

/// Check if a given path is absolute
fn is_absolute_path(path: &Path) -> bool {
    path.starts_with("/")
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_relative_paths() {
        let paths = vec!["abc"];
        for path in paths {
            assert!(is_relative_path(Path::new(path)));
        }
    }
    
    #[test]
    fn test_absolute_paths() {
        let paths = vec!["/xyz", "/foo/bar"];
        for path in paths {
            assert!(is_absolute_path(Path::new(path)));
        }
    }

    #[test]
    #[cfg(windows)]
    fn test_windows_path_behavior() {
      todo!()
    }
}
```

Reply:
when I went to make this change I got confused and noticed I couldn't reproduce my problem anymore, so just reverted for the change in this PR, if I run into Windows issues in the future I'll send separate PR